qlib/TableMapper.qm

```diff
@@ -663,13 +665,14 @@ on_error table_mapper.rollback();
             # copy all runtime mappings to the output hash
             map dh{$1.key} = m_runtime{$1.value}, rconsth.pairIterator();
 
-            int size = rec.firstValue().size();
-            map mapFieldIntern(\dh, $1, rec, size), mapd.keyIterator();
-            count += size;
+            map mapFieldIntern(\dh, $1, rec, True, rec_list_size), mapd.keyIterator();
+            count += rec_list_size;
 
             # map record data to get the keys for the buffer
             if (!hbuf) {
                 # copy all constant mappings to the output hash
+                # FIXME why copying const dat here again even if it is
```

Comment:
`consth` is not processed in `mapFieldIntern()` - is is removed from `mapd` in `Mapper::checkMap()` as a performance optimization - we don't want to process the constant values for each row when they will only be included once in the bulk DML call